Use all man pages inside `man` directory

gettalong opened the pull request:
The `man` command allows placing manual pages either directly inside a `man` directory or inside the `man1`, `man2`, ... directories. Also use the files in the nested directories.

bitboxer commented:
Yes, this would fix the problem with your gem, but I am not sure I want to go this path. This would make this gem change from the behavior that the `gem-man` gem uses. Every gem that uses the convention here will be incompatible with it. And I don't see any advantage for doing this. You will need to explain me here why you are sure it is a good idea to break compatibility with a gem that has 32k downloads and is currently the defacto standard in dealing with man pages in ruby gems.

gettalong commented:
The behaviour would change, yes, but it would be compatible to `gem-man` because it not just searches in the `man/` directory but all its subdirectories, too. In this way, your gem would instantly find *more* man pages than before (at least in my installation it did...).

For what's worth, I didn't know that there was a convention for rubygems that only files directly in the `man/` directory are to be considered, in stark contrast to what the `man` command itself does. When I added the man page to kramdown back in February 2010, I just used the, for me, most natural path. And also the one I used for webgen earlier, in February 2007.

As for being compatible with `gem-man`: Your `manpages` gem is **much** more useful than it due to not having to think whether one has to use `man` or `gem man`! :smile:

So: your project, your decision - above are just my 2c. However you decide thanks for this project! :+1:

bitboxer commented:
Yes, the regexp in the `gem-man` looks like it might work, but actually the gem runs into an exception when trying to read the man file 😉 . But okay, you convinced me. I will merge this in. One little thing though: can you make travis happy? I added the codestyle rules of the bundler project to make it easier to integrate this into the rubygem in a far far future. And the rspec throws a couple of errors, too.

gettalong commented:
I have updated the specs and fixed everything. One note though: the test "handles permission problems gracefully" on installation fails for me locally due to ordering differences.

bitboxer commented:
Thanks for your fix. I created #37 for the flaky test and will investigate it there.

bitboxer commented:
Just a small heads up: I fixed the flaky test and released version 0.6. Now this gem is compatible with hexapdf. Just tried a `man hexapdf` and it worked like a charm.

gettalong commented:
Thanks! Already installed and works great! :smile: :+1:
